### DISCUSSION TITLE
Tab method

### DISCUSSION THREAD

**Discussion by v479038280** [zauberzeug/nicegui#1367] [Ideas / Feature Requests]
If the ui.tab() could be cloesed or created danamicaly like a new page.

**Comment by rodja** [comment]
Please be more specific. What would you like to do? Can you share some code? We really try to respond to every discussion/issue and it would help if you take some time to formulate your idea.

**Reply by v479038280** [reply]
Just like the tab in our browser ,It could be closed(with the X button in the red circle) and created dynamically in some way ( e.g. with the `ui.open(tab='The tab we want to open')` )
![微信截图_20230808162653](https://github.com/zauberzeug/nicegui/assets/25096045/d5169573-7a95-4041-ab28-a082ea1dc8f0)

Is this picture  can clearly express my thoughts?

**Comment by rodja** [comment]
This may be already archivable with the existing tools. But it will require some vue3/slots knowledge. Anybody interested in implementing / demoing?

**Reply by maheshs11** [reply]
#1270 maybe this way possible. 
```py
with ui.tabs() as tabs:
   with ui.tab('tabone') as tabone:
         ui.icon('close').on('click',lambda: setattr(tabone, 'visibility', False)) 
```

**Reply by v479038280** [reply]
Could we make it a component ready to use?

**Reply by maheshs11** [reply]
Yes. you can make it, but i am not so familiar with nicegui internals.
